<?xml version="1.0"?>
<launch>
    <arg name="compress" default="true"/>
    <arg name="undistort" default="false"/>
    <arg name="debug" default="false"/>
    <arg name="get_images" default="false"/>
    
    <arg name="config" default="config.yaml"/>
    
    <node name="bringup" pkg="insta360_ros_driver" type="insta360_ros_driver"/>

    <group if="$(arg compress)">
        <rosparam file="$(find insta360_ros_driver)/config/$(arg config)" command="load"/>
        <param name="undistort" value="$(arg undistort)"/>

        <node unless="$(arg debug)" name="live_processing" pkg="insta360_ros_driver" type="live_processing.py"/>

        <node if="$(arg debug)" name="live_processing" pkg="insta360_ros_driver" type="live_processing.py" output="screen" launch-prefix="xterm -e"/>
        
        <node if="$(arg debug)" name="rqt_image_view" pkg="rqt_image_view" type="rqt_image_view" args="/back_camera_image/compressed"/>
        <node if="$(arg debug)" name="rostopic_hz" pkg="insta360_ros_driver" type="rostopic_hz.sh" output="screen" launch-prefix="xterm -e"/>

        <node if="$(arg get_images)" name="get_images" pkg="insta360_ros_driver" type="get_images.py" output="screen">
            <param name="topic" value="/back_camera_image/compressed"/>
        </node>
    </group>

    <group unless="$(arg compress)">
        <node if="$(arg debug)" name="rqt_image_view" pkg="rqt_image_view" type="rqt_image_view" args="/insta_image_yuv"/>
    
        <node if="$(arg debug)" name="rostopic_hz" pkg="insta360_ros_driver" type="rostopic_hz_raw.sh" output="screen" launch-prefix="xterm -e"/>
    </group>
</launch>